### PR TITLE
feat: added priceDetail functionality

### LINKED ISSUE #23: adding price detail functionality
adding functionality for calculating bill

### PR DESCRIPTION
**Pull request by mariyasada**
adding functionality to calculate the total bill of  products
and also adding functionality for removing items from the cart
please check the calculating process if it works properly or not

https://user-images.githubusercontent.com/77275100/159115964-78868d16-6e37-46d4-9756-7f7954e3e9f5.mp4

### CODE REVIEW

File: src/context/Reducer/CartItemReducer.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
 export const CartItemReducer=(state,action)=>{
+    console.log(state,"Hello");
```
Comment (Bhakti199) [inline src/context/Reducer/CartItemReducer.js]: remove extra comment

File: src/components/TextOnlyCard/TextOnlyCardWithPrice.jsx [hunk 2]
```diff
@@ -1,57 +1,75 @@
...
+  //  calculating total sellingPrice
+  const totalSellingPrice = [...cartItem].reduce(
+    (acc, currentItem) => currentItem.qty * currentItem.sellingPrice + acc,
+    0
+  );
+  // calculating Bag discount (original-selling)
+  const BagDiscount = totalOriginalPrice - totalSellingPrice;
+
+  //  billing
+  const GrandTotal = totalSellingPrice + DeliveryCharges;
```
Comment (Bhakti199) [inline src/components/TextOnlyCard/TextOnlyCardWithPrice.jsx]: Functions are defined  well

**Review by Bhakti199 — APPROVED** [review]
This code looks fine to me,
Nicely written and structured properly. 
This PR is good to merge.

File: src/context/Reducer/CartItemReducer.js (hunk 1 above)
Comment (mariyasada) [reply src/context/Reducer/CartItemReducer.js]: will do

**Review by harshalini — APPROVED** [review]
Code is good, you can merge it

File: src/components/TextOnlyCard/TextOnlyCardWithPrice.jsx [hunk 3]
```diff
@@ -1,57 +1,75 @@
...
+  const totalOriginalPrice = [...cartItem].reduce(
+    (acc, currentItem) => currentItem.qty * currentItem.originalPrice + acc,
+    0
+  );
+  //  calculating total sellingPrice
+  const totalSellingPrice = [...cartItem].reduce(
+    (acc, currentItem) => currentItem.qty * currentItem.sellingPrice + acc,
+    0
+  );
+  // calculating Bag discount (original-selling)
```
Comment (JantuDeb) [inline src/components/TextOnlyCard/TextOnlyCardWithPrice.jsx]: You don't need to destructure `cartItem` here, `reduce does not mutate original Array object

**Review by JantuDeb — APPROVED** [review]
This looks fine

File: src/components/TextOnlyCard/TextOnlyCardWithPrice.jsx (hunk 3 above)
Comment (mariyasada) [reply src/components/TextOnlyCard/TextOnlyCardWithPrice.jsx]: okay

**Pull request merged**